fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsCollectJobFactory.java

```diff
@@ -109,8 +110,24 @@ public static StatisticsCollectJob buildStatisticsCollectJob(Database db, Table
         LOG.debug("statistics job work on table: {}, type: {}", table.getName(), analyzeType.name());
         if (analyzeType.equals(StatsConstants.AnalyzeType.SAMPLE)) {
             if (partitionIdList == null) {
-                partitionIdList = table.getPartitions().stream().filter(Partition::hasData)
-                        .map(Partition::getId).collect(Collectors.toList());
+                Collection<Partition> partitions = table.getPartitions();
+                BasicStatsMeta basicStatsMeta = GlobalStateMgr.getCurrentState().getAnalyzeMgr()
+                        .getTableBasicStatsMeta(table.getId());
+
+                if (basicStatsMeta != null) {
+                    partitionIdList = partitions.stream()
+                            .filter(partition -> {
+                                LocalDateTime partitionUpdateTime = StatisticUtils.getPartitionLastUpdateTime(partition);
+                                return basicStatsMeta.getUpdateTime().isBefore(partitionUpdateTime) && partition.hasData();
```

Comment:
is it correct ? the `basicStatsMeta.getUpdateTime()` is the update time of last job, but it can be applied on partial partitions of columns, which might not same as this job ?

Reply:
currently our auto collecting statistics will collect all partitions and all columns.  I think this is equivalent to partition-level column update time